chain/x/oracle/keeper/reporter.go

```diff
@@ -33,3 +33,22 @@ func (k Keeper) RemoveReporter(ctx sdk.Context, val sdk.ValAddress, addr sdk.Acc
 	ctx.KVStore(k.storeKey).Delete(types.ReporterStoreKey(val, addr))
 	return nil
 }
+
+// GetAllReportersOfValidator returns the reporter list of the given validator.
+func (k Keeper) GetAllReportersOfValidator(ctx sdk.Context, val sdk.ValAddress) (reporters []sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	storeKeyPrefix := append(types.ReporterStoreKeyPrefix, val.Bytes()...)
+	iterator := sdk.KVStorePrefixIterator(store, storeKeyPrefix)
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		key := iterator.Key()
+		reporterBytes := key[21:]
+		reporterAddress := sdk.AccAddress(reporterBytes)
+		reporters = append(reporters, reporterAddress)
+	}
+
+	// Appends self reporter of validator to the list
+	selfReporter := sdk.AccAddress(val)
```

Comment:
Can we add self first? I think having self as the first element makes more sense.